include/openthread/link_metrics.h

```diff
@@ -0,0 +1,167 @@
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file defines the OpenThread Link Metrics API.
+ */
+
+#ifndef LINK_METRICS_H_
+#define LINK_METRICS_H_
+
+#include "openthread-core-config.h"
+
+#include <openthread/ip6.h>
+#include <openthread/message.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup linkmetrics   Link Metrics
+ *
+ * @brief
+ *   This module includes functions that control Link Metrics protocol.
+ *
+ * @{
+ *
+ */
+
+/**
+ * This enumeration defines Link Metrics ID.
+ *
+ */
+typedef enum otLinkMetricsId
+{
+    OT_LINK_METRICS_PDU_COUNT = 0, ///< Layer 2 PDUs Received
+    OT_LINK_METRICS_LQI       = 1, ///< Layer 2 LQI
+    OT_LINK_METRICS_MARGIN    = 2, ///< Link Margin - RSSI margin above noise floor
+    OT_LINK_METRICS_RSSI      = 3, ///< RSSI
+} otLinkMetricsId;
+
+/**
+ * This enumeration defines Link Metrics Type/Average Enum.
+ *
+ */
+typedef enum otLinkMetricsTypeEnum
+{
+    OT_LINK_METRICS_TYPE_COUNT       = 0, ///< Count/summation
+    OT_LINK_METRICS_TYPE_EXPONENTIAL = 1, ///< Exponential Moving Average
+    OT_LINK_METRICS_TYPE_RESERVED    = 7, ///< 2 - 7 are reserved
+} otLinkMetricsTypeEnum;
+
+/*
+ * This structure represents what metrics are specified to query.
+ *
+ */
+typedef struct otLinkMetrics
+{
+    bool mPduCount : 1;
+    bool mLqi : 1;
+    bool mLinkMargin : 1;
+    bool mRssi : 1;
+} otLinkMetrics;
+
+/*
+ * This structure represents the result (value) for a Link Metrics query.
+ *
+ */
+typedef struct otLinkMetricsValues
+{
+    otLinkMetrics mMetrics; ///< Specifies which metrcis values are present/included.
+
+    uint32_t mPduCountValue;
+    uint8_t  mLqiValue;
+    int8_t   mLinkMarginValue;
```

Comment:
Shouldn't this be `uint8_t`? Negative link margin doesn't make much sense.

When testing in nRF52840 I observe something like this:
```
> linkmetrics query fe80:0:0:0:40e7:903:9104:f9f5 single m
Done
> Received Link Metrics Report from: fe80:0:0:0:40e7:903:9104:f9f5
 - Margin: -98 (Exponential Moving Average)
```